holopy/inference/interface.py

```diff
@@ -74,36 +74,59 @@ def validate_strategy(strategy, operation):
     return strategy
 
 
-def make_default_model(base_scatterer, fitting_parameters):
+def make_default_model(base_scatterer, fitting_parameters=None):
     if fitting_parameters is None:
-        fitting_parameters = base_scatterer.parameters.keys()
+        fitting_parameters = list(base_scatterer.parameters.keys())
     scatterer = parameterize_scatterer(base_scatterer, fitting_parameters)
     alpha_prior = Uniform(0, 1, guess=0.7, name='alpha')
     return AlphaModel(scatterer, noise_sd=1, alpha=alpha_prior)
 
 
 def parameterize_scatterer(base_scatterer, fitting_parameters):
-    parameters = base_scatterer.guess.parameters
+    if isinstance(fitting_parameters, str):
+        fitting_parameters = [fitting_parameters]
+    parameters = base_scatterer.parameters
     variable_parameters = {par_name: make_uniform(parameters, par_name)
-                           for par_name in rename_xyz(fitting_parameters)}
+                           for par_name in fitting_parameters}
     parameters.update(variable_parameters)
-    return type(base_scatterer)(**_interpret_parameters(parameters, True))
-
-
-def rename_xyz(parameters_list):
-    for i, key in enumerate(['x', 'y', 'z']):
-        new_key = 'center.{}'.format(i)
-        if key in parameters_list:
-            loc = parameters_list.index(key)
-            parameters_list[loc] = new_key
-    return parameters_list
+    for i, key in enumerate(COORD_KEYS):
+        if isinstance(base_scatterer, Scatterers):
+            for j in range(len(base_scatterer.scatterers)):
+                replace_center(parameters, "{}:{}".format(j, key))
+        else:
+            replace_center(parameters, key)
+    return base_scatterer.from_parameters(parameters)
+
+
+def replace_center(parameters, key):
+    index = COORD_KEYS.index(key[-1])
+    prefix = key[:-1]
+    try:
+        val = parameters.pop(key)
+    except KeyError:
+        pass
+    else:
+        parameters[prefix + 'center'][index] = val
```

Comment:
I'm a little confused as to what this code does, esp. since you said in the PR that
> scatterer.parameters used to give a flat dictionary, e.g. {n.real: Uniform(1.58, 1.60), n.imag: 0.001, 'center.0': 10, 'center.1':10, 'center.2':Uniform(0, 100)}. After this PR it matches the initial arguments to call the scatterer: {'n': ComplexPrior(Uniform(1.58, 1.60), 0.001), 'center': (10, 10, Uniform(0, 100))} .

Can you let me know what I'm missing, either in the comments or when we meet?

Reply:
Oh... is this because of `Spheres` vs `Sphere`?

Reply:
Yeah this is to handle the case of Scatterers objects, which might have '2:y' as a user-specified fitting parameter that needs to replaced with '2:center'[1] since that's how it would appear in Spheres.parameters.